### ISSUE TITLE
As a Canvas Course participant I would like to be able to 

### ISSUE THREAD

**Issue by ekellagher** [CuBoulder/express#2016]
I am creating self-paced training in Canvas for Web Express clients. Clients may self enroll in the classes and part of this enrollment process is to receive a test training site to complete the course assignments. We train any where from 600 - 700 people a year so I would like to automate the test site allocation process It would be nice if there was a process where clients could automatically receive a test site when they register for a course. They would actually only need one training site a month since they could use the training site for other courses.

Right now I can create a form for them to fill out when they register but there will be a lag between registration and being able to start the course until they get a site.

Thanks

**Comment by kreynen** [comment]
Because the unlaunched sites on www-test are "self destructing" and not public, I have no issue with assigning the first non-SRC staff user that logs in as the SO.  We could just send them to a URL like http://www.colorado.edu/webcentral/get-training-site that would redirect them to an available site on www-test.  As long as there is a site available, they'd just need to log in.  If there where no sites available, we'd just say something like "Sorry.  Web Express is very popular and we've allocated all of the available sites, but don't worry we're busy making right now.  Please check back in 5 minutes and we should have a site ready for you.  If not, please contact websupport@colorado.edu"

As long as we are using LDAP, we still need to deal with the window of time where the user was sent to the login page of the site, but haven't actually logged.  Otherwise we could send 2 people to the same sites.  The first one to successfully login is the SO.  The other is just an authenticated user.  

One option to address this would be to keep track of the P1s of the last 2 sites that redirect has sent users to.  

We'd need to add some logic similar to what we use for the first invite to update the site's status too.

There are other ways we could handle this, but this would be a pretty simple way to hand out test sites by adding just a few lines of code. 

**Developer Notes:**  
In the same user login hook as https://github.com/CuBoulder/express/blob/dev/modules/features/cu_ldap/cu_ldap.module#L70 or the same hook somewhere else...

- check to see if the site is on www-test using env variables
- check to see if this is the first user logging in by checking for values in the access column of the users table.  
-  confirm it is not SRC staff 

If all of those are TRUE, make that user a site owner and change the site's status.
